### PR TITLE
Add support for relative dates to DateTime fields

### PR DESCRIPTION
**Pull request by cpennington**

### CODE REVIEW

File: xblock/fields.py [hunk 1]
```diff
@@ -965,29 +965,42 @@ def from_json(self, value):
...
+
             try:
                 value = dateutil.parser.parse(value)
             except (TypeError, ValueError):
                 raise ValueError("Could not parse {} as a date".format(value))
 
-        if not isinstance(value, datetime.datetime):
+        # Interpret raw numbers as a relative dates
+        if isinstance(value, (int, float)):
+            return datetime.timedelta(seconds=value)
```
Comment (jmyatt) [inline xblock/fields.py]: this line might be better written as:
`value = datetime.timedelta(seconds=value)`

That way the general flow is 1) interpret/transform `value`, 2) validate `value`'s type 3) fix up and return `value`.

File: xblock/fields.py [hunk 2]
```diff
@@ -965,29 +965,42 @@ def from_json(self, value):
...
         if isinstance(value, datetime.datetime):
             return value.strftime(self.DATETIME_FORMAT)
+
+        if isinstance(value, datetime.timedelta):
+            return value.total_seconds()
+
         if value is None:
             return None
+
         raise TypeError("Value stored must be a datetime object, not {}".format(type(value)))
```
Comment (jmyatt) [inline xblock/fields.py]: maybe add timedelta as a possible stored value in this message?

File: xblock/fields.py (hunk 1 above)
Comment (cpennington) [reply xblock/fields.py]: Oh, yeah, the `return` was a brainfart.

**Comment by bradenmacdonald** [comment]
Nice :)

**Pull request merged**